tests/test_ioc_system.py

```diff
@@ -337,18 +337,28 @@ async def test_bobfiles_created(mocked_panda_standard_responses):
     # Wait for the files to be created in the subprocess.
     await asyncio.sleep(1)
 
-    old_files = os.listdir(BOBFILE_DIR)
-    for file in old_files:
+    old_files = [
+        file_path
+        for file_path in Path.iterdir(BOBFILE_DIR)
+        if file_path.suffix == ".bob"
+    ]
+    for file_path in old_files:
         assert (
-            Path(bobfile_temp_dir / file)
+            Path(bobfile_temp_dir / file_path.name)
             .read_text()
-            .replace(test_prefix, "TEST-PREFIX")
-            == (BOBFILE_DIR / file).read_text()
+            .replace(test_prefix, TEST_PREFIX)
+            == file_path.read_text()
         )
 
     # And check that the same number of files are created
-    new_files = os.listdir(bobfile_temp_dir)
-    assert len(old_files) == len(new_files)
+    new_files = list(Path.iterdir(bobfile_temp_dir))
+    if len(old_files) != len(new_files):
+        raise AssertionError(
```

Comment:
This error message should probably be printed for the other `assert` statement in this test too. 

Perhaps do the whole test inside a `try-except AssertionError` and then customise the error message there? Saves repeating the message twice.

Reply:
I have a minor preference for defining the error output a string and using it on both asserts if that's cool with you

Reply:
Sure go ahead